### PR TITLE
add `addIndex` & `dropIndex` @ `AlterTableBuilder`

### PR DESCRIPTION
**Pull request by Gaspero**
closes https://github.com/kysely-org/kysely/issues/507
closes https://github.com/kysely-org/kysely/issues/508 

- [x]  implement
- [x]  test
- [x]  document

### CODE REVIEW

File: src/schema/alter-table-builder.ts [hunk 1]
```diff
@@ -224,6 +228,25 @@ export class AlterTableBuilder implements ColumnAlteringInterface {
...
   }
 
+  addIndex(indexName: string): AlterTableAddIndexBuilder {
+    return new AlterTableAddIndexBuilder({
+      ...this.#props,
+      indexBuilder: new AddIndexBuilder({
+        ...this.#props,
+        node: AddIndexNode.create({ name: IdentifierNode.create(indexName) }),
+      }),
+    })
```
Comment (igalklebanov) [inline src/schema/alter-table-builder.ts:238]: I don't understand why both `AlterTableAddIndexBuilder` and `AddIndexBuilder` exist. Mind explaining? 😕

File: src/operation-node/add-index-node.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * @internal
+ */
+export const AddIndexNode = freeze({
+  is(node: OperationNode): node is AddIndexNode {
+    return node.kind === 'AddIndexNode'
+  },
+
+  create(index: Omit<AddIndexNode, 'kind'>): AddIndexNode {
```
Comment (igalklebanov) [inline src/operation-node/add-index-node.ts:25]: Is there ever a situation when it's created with anything but a name at first?
If not, accept a single argument `name` here.

**Review by igalklebanov — COMMENTED** [review]
Hey 👋 

Thank you! 💪 

Looks good for the most part, a few comments and suggestions..

File: test/node/src/schema.test.ts [hunk 3]
```diff
@@ -1890,6 +1890,145 @@ for (const dialect of DIALECTS) {
             await builder.execute()
           })
         })
+
+        describe('add index', () => {
```
[the remark this answers is not in the document]
Comment (igalklebanov) [reply test/node/src/schema.test.ts:1894]: We should break this test file into multiple files, but it is not in scope for this PR.

**Comment by harrytran998** [comment]
Hi @igalklebanov, can you take the rest of these features? Bcs currently, this in the only way to create an index on a table as I know with Kysely.

Thank you!

**Comment by Gaspero** [comment]
Hey everyone!

Sorry for the delay in applying suggestions. I hope I will manage to fix everything util the end of August as I am currently stuck with my pet project.

> Bcs currently, this in the only way to create an index on a table as I know with Kysely. 

 @harrytran998 AFAIK in MySQL you could use [create index statement](https://dev.mysql.com/doc/refman/8.0/en/create-index.html).
For reference see tests: https://github.com/kysely-org/kysely/blob/56107c59fb6e553846e8687a167a3f5615f9b613/test/node/src/schema.test.ts#L934

**Comment by koskimas** [comment]
This PR seems to be dead. Closing.

**Comment by Gaspero** [comment]
Hey everyone!

Sorry, I kinda lost track of time. I've completed the milestone of the project I was working at. I hope I will be able to finish implementation of this features now.

I've made a separate branch where I applied @igalklebanov suggestions https://github.com/Gaspero/kysely/tree/add-and-remove-index-in-alter-table-v2

Don't you mind if I open a new PR?

**Pull request closed without merging**